activestorage/app/models/active_storage/record.rb

```diff
@@ -2,6 +2,10 @@
 
 class ActiveStorage::Record < ActiveRecord::Base # :nodoc:
   self.abstract_class = true
+
+  def self.full_table_name_prefix
+    table_name_prefix + super
```

Comment:
I had to add this here, but I believe this is a global rails/ActiveRecord issue, for any application that wans to have "Base classes" that respects ActiveRecord prefixes

I believe this should be investigated in another PR. for now this changes fixes the issue **and is not breaking for existing applications**

Reply:
I believe since ActiveStorage is an engine, `ActiveStorage.table_name_prefix` is set
![image](https://github.com/rails/rails/assets/1371733/c1cd46dd-24ae-493d-8726-70de299eee02)

which makes
```ruby
def full_table_name_prefix # :nodoc:
 (module_parents.detect { |p| p.respond_to?(:table_name_prefix) } || self).table_name_prefix
end
```

returns only `active_storage_` and discarding completely ActiveRecord prefix config

and for Suffix
```ruby
def full_table_name_suffix # :nodoc:
(module_parents.detect { |p| p.respond_to?(:table_name_suffix) } || self).table_name_suffix
end
```

the `detect` call will be nil, hence relying on `self` and since `self` is the AS model, then it will be `ActiveStorage::Blob.table_name_suffix` 

==> I believe we fully understand what's happening now.

Reply:
I think that this is a global issue for Engines issue that use `isolate_namespace`